### PR TITLE
Test the query builder for a salary range.

### PR DESCRIPTION
**Pull request by gmackay**

### CODE REVIEW

File: vcm/src/test/resources/repository/SalaryRangeSelectValuesQuery.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+SELECT * FROM vacancies WHERE public_opening_date IS NOT NULL AND public_opening_date <= :now AND (point(:searchFromLongitudeValue, :searchFromLatitudeValue) <@> point(longitude, latitude)) < :distance AND salary_max >= :salary_min
```
Comment (reachsaurabh) [inline vcm/src/test/resources/repository/SalaryRangeSelectValuesQuery.sql]: Why does test have sql files , I would expect to test the actual source code class and assert , its like testing a test source

Comment (seamussingh) [reply vcm/src/test/resources/repository/SalaryRangeSelectValuesQuery.sql]: The SearchQueryBuilder is responsible for generating dynamic sql based on the state of SearchParameters and what is found in terms of values supplied.  This class produces sql that is then used in JPA Native query to create a query object.

The .sql files in test/resources are expected results given the a given combination of search parameters. The test class simply compares the actual sql generated against the expected result which are stored as sql snippets in test/resources.

File: vcm/src/test/java/uk/gov/cshr/vcm/repository/SearchQueryBuilderTest.java [hunk 2]
```diff
@@ -76,6 +75,23 @@ public void buildSelectValuesQuery_onlyLocationSupplied() throws IOException {
         doSelectValuesQueryTest(buildParameters(BRISTOL, null, null), LOCATION_SELECT_VALUES_QUERY);
     }
 
+    @Test
+    public void buildSelectValuesQuery_SalaryRange() throws IOException {
```
Comment (seamussingh) [inline vcm/src/test/java/uk/gov/cshr/vcm/repository/SearchQueryBuilderTest.java]: Are there additional tests that cover the other combinations of building the salary range query, this appears only to address "salary_max >= :salary_min".  Is that the only possible condition in the where clause?

File: vcm/src/test/resources/repository/SalaryRangeSelectValuesQuery.sql (hunk 1 above)
Comment (gmackay) [reply vcm/src/test/resources/repository/SalaryRangeSelectValuesQuery.sql]: I wonder if it there might be a better way of checking the query builder is working as expected?  One problem I noticed was simply by adding an extra bit of SQL every test failed and eight sql files then needed updating.

What I did instead was change the equals to a contains test.  But it still seems maintenance heavy.

Is a more behaviour driven approach not adequate here?

File: vcm/src/test/java/uk/gov/cshr/vcm/repository/SearchQueryBuilderTest.java (hunk 2 above)
Comment (gmackay) [reply vcm/src/test/java/uk/gov/cshr/vcm/repository/SearchQueryBuilderTest.java]: I have added some.

**Comment by seamussingh** [comment]
This seems to have been dealt with other changes.

**Pull request closed without merging**